File: julia/14.jl
function collatzLen(n)
	chainLength = 0

	while n !== 1
		if n%2 === 0
			n = div(n,2)
		else
			n = 3n + 1
		end
		chainLength += 1
	end

	return chainLength
end

maxLen = 0
maxNum = 0
for i = 1:1_000_000
	if collatzLen(i) > maxLen
		global maxLen = collatzLen(i)
		global maxNum = i
	end
end

println(maxNum)
# 837799

File: c/14.c
#include <stdio.h>

int collatzLen(long n) {
  int chainLen = 0;

  while (n != 1) {
    if (n % 2 == 0) {
      n /= 2;
    } else {
      n = 3 * n + 1;
    }

    chainLen += 1;
  }

  return chainLen;
}

int main() {
  int maxLen = 0;
  int maxNum = 0;
  for (long i = 1; i <= 1e6; i++) {
    if (collatzLen(i) > maxLen) {
      maxLen = collatzLen(i);
      maxNum = i;
    }
  }

  printf("%d\n", maxNum);
  return 0;
}
// 837799
